savant/deepstream/utils/object.py

```diff
@@ -154,32 +139,147 @@ def nvds_generate_obj_uid(
 
-    if nvds_get_obj_selection_type(nvds_obj_meta) == ObjectSelectionType.ROTATED_BBOX:
-        return NvDsRBBox(nvds_obj_meta)
+    if bbox_type == ObjectSelectionType.ROTATED_BBOX:
+        return get_rotated_bbox_from_obj_meta(nvds_obj_meta)
 
     raise IncorrectSelectionType('Unsupported object selection type.')
 
 
+def nvds_set_obj_bbox(
+    batch_meta: pyds.NvDsBatchMeta,
+    obj_meta: pyds.NvDsObjectMeta,
+    bbox: Union[BBox, RBBox],
+) -> None:
+    """Set bbox values and object selection type for specified object meta
+    into Deepstream meta structures."""
+    if isinstance(bbox, BBox):
+        set_aligned_bbox_for_obj_meta(obj_meta, bbox)
+        nvds_set_obj_selection_type(
+            obj_meta,
+            ObjectSelectionType.REGULAR_BBOX,
+        )
+
+    elif isinstance(bbox, RBBox):
+        set_rotated_bbox_for_obj_meta(batch_meta, obj_meta, bbox)
+        nvds_set_obj_selection_type(
+            obj_meta,
+            ObjectSelectionType.ROTATED_BBOX,
+        )
+
+
+def set_aligned_bbox_for_obj_meta(obj_meta: pyds.NvDsObjectMeta, bbox: BBox) -> None:
+    """Set aligned bbox values for specified object meta
+    into Deepstream meta structures."""
+    bbox_coords = obj_meta.detector_bbox_info.org_bbox_coords
+    bbox_coords.left = bbox.left
+    bbox_coords.top = bbox.top
+    bbox_coords.width = bbox.width
+    bbox_coords.height = bbox.height
+
+    rect_params = obj_meta.rect_params
+    rect_params.left = bbox.left
+    rect_params.top = bbox.top
+    rect_params.width = bbox.width
+    rect_params.height = bbox.height
+
+
+def set_rotated_bbox_for_obj_meta(
+    batch_meta: pyds.NvDsBatchMeta, obj_meta: pyds.NvDsObjectMeta, bbox: RBBox
+) -> None:
+    """Set rotated bbox values for specified object meta
+    into Deepstream meta structures."""
+    rbbox_coords = NvRBboxCoords()
+    rbbox_coords.x_center = bbox.xc
+    rbbox_coords.y_center = bbox.yc
+    rbbox_coords.width = bbox.width
+    rbbox_coords.height = bbox.height
+    rbbox_coords.angle = bbox.angle
+    add_rbbox_to_object_meta(batch_meta, obj_meta, rbbox_coords)
+
+
+def nvds_upd_obj_bbox(obj_meta: pyds.NvDsObjectMeta, bbox: Union[BBox, RBBox]):
+    """Update bbox values for specified object meta
+    into Deepstream meta structures."""
+    if isinstance(bbox, BBox):
+        set_aligned_bbox_for_obj_meta(obj_meta, bbox)
+
+    elif isinstance(bbox, RBBox):
+        update_rotated_bbox_for_obj_meta(obj_meta, bbox)
+
+
+def update_rotated_bbox_for_obj_meta(
+    obj_meta: pyds.NvDsObjectMeta, bbox: RBBox
+) -> None:
+    rbbox = get_rbbox(obj_meta)
+    if rbbox is None:
+        raise MetaValueError(
+            'No rotated bounding box found in Deepstream meta information.'
+        )
+    rbbox.x_center = bbox.xc
+    rbbox.y_center = bbox.yc
+    rbbox.width = bbox.width
+    rbbox.height = bbox.height
+    rbbox.angle = bbox.angle
+
+
+def get_aligned_bbox_from_obj_meta(obj_meta: pyds.NvDsObjectMeta) -> BBox:
```

Comment:
I think it improves readability.

Reply:
For me personally, this only makes it worse to read, because it requires you to go into a function to understand what's going on, and the function just assigns values. Also, you replaced the passing of arguments by using positional arguments and you say that reducing readability is less important because it adds performance. And in this case you reduce performance, but say it increases readability. If we are looking for performance, then we should definitely get rid of functions consisting of several lines and only called in one place.